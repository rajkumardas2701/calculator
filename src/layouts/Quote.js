import NavBar from './NavBar';

const Quote = () => (
  <div className="AppBody">
    <NavBar />
    <h3 id="quote">
      Mathematics is not about numbers, equations, computations,
      or algorithms: it is about understanding.
    </h3>
    <h3 id="author">
      â€“William Paul Thurston
    </h3>
  </div>
);

export default Quote;
