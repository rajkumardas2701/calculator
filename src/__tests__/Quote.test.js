import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Quote from '../layouts/Quote';

afterEach(cleanup);

it('should display the quote text', () => {
  const { getByTestId } = render(<Quote />);
  expect(getByTestId('quote-test')).toHaveTextContent(/Mathematics is not about numbers/);
});

it('renders correctly', () => {
  const quote = renderer.create(<Quote />).toJSON();
  expect(quote).toMatchSnapshot();
});
