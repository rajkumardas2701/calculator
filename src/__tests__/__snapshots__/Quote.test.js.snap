// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="AppBody"
>
  <nav>
    <h1
      data-testid="navBar-header"
    >
      Math Magicians
    </h1>
    <div
      className="rightNav"
      data-testid="navBar-test"
    >
      <div>
        <a
          href="/"
        >
          Home
        </a>
      </div>
      <div>
        <a
          href="/calculator"
        >
          Calculator
        </a>
      </div>
      <div>
        <a
          href="/quote"
        >
          Quotes
        </a>
      </div>
    </div>
  </nav>
  <h3
    data-testid="quote-test"
    id="quote"
  >
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding.
  </h3>
  <h3
    id="author"
  >
    â€“William Paul Thurston
  </h3>
</div>
`;
